### DISCUSSION TITLE
Database Notification notifiable_type?

### DISCUSSION THREAD

**Discussion by RPotkowaVegasystems** [filamentphp/filament#8424] [Help]
My Database Notifications are not set to notifiable_type User, i use a another Model. How can i change the view or the query to use my Model instead the User Model?

**Comment by danharrin** [comment]
Is the model related to the user in some way?

**Comment by RPotkowaVegasystems** [comment]
Yes. 

User (BelongsTo) -> MyModel
MyModel(HasMany) -> User

**Comment by RPotkowaVegasystems** [comment]
I see that the component "Filament\Notifications\Livewire\DatabaseNotifications.php" -> getNotificationsQuery() is responsible for the results. Is it maybe possible to replace it or use you own DatabaseNotifications Class? It would be fine, if your also could provide in the panel configuration "$panel->databaseNotifications(MyOwnComponent::class)" you own class.

**Reply by danharrin** [reply]
Could you maybe swap the class out in the container?

**Reply by RPotkowaVegasystems** [reply]
How? I added in my AppServiceProvider
```
    public function register(): void
    {
        $this->app->bind(DatabaseNotifications::class, CustomDatabaseNotifications::class);

    }
```

But it not worked.

**Comment by danharrin** [comment]
Try `Filament\Livewire\DatabaseNotifications` rather than `Filament\Notifications\Livewire\DatabaseNotifications` if you're inside a panel

**Comment by RPotkowaVegasystems** [comment]
Ok. Thats also not worked. But i went an another approach.

1. I created a Livewire Component thats extends your "Filament\Livewire\DatabaseNotifications" with my modified "getNotificationsQuery()".
2. Created a view and put the Livewire Component in it.
3. Registered the view in the "panels::global-search.after"
4. I disabled the "->databaseNotifications()" in the panel conf.
